### DISCUSSION TITLE
windows bat file not running in Github actions

### DISCUSSION THREAD

**Discussion by epneo-banuprakash** [community/community#33225] [Actions]
hi i am trying to run  bat file in windows self hosted runner, but it gives error: cmd command not found.
my yaml file like below

name: esp-idf
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]  
jobs:
  build:
    runs-on: [self-hosted, Windows, X64]
    env:
       
      PATH: C:\Espressif\python_env\idf4.4_py3.8_env\Scripts\;C:\Espressif\tools\idf-git\2.34.2\cmd\;C:\Espressif;C:\Espressif\tools\idf-python\3.8.7\;C:\Espressif\tools\idf-git\2.34.2\cmd\;C:\Espressif\frameworks\esp-idf-v4.4.1\components\esptool_py\esptool;C:\Espressif\frameworks\esp-idf-v4.4.1\components\app_update;C:\Espressif\frameworks\esp-idf-v4.4.1\components\espcoredump;C:\Espressif\frameworks\esp-idf-v4.4.1\components\partition_table;C:\Espressif\tools\xtensa-esp32-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32-elf\bin;C:\Espressif\tools\xtensa-esp32s2-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s2-elf\bin;C:\Espressif\tools\xtensa-esp32s3-elf\esp-2021r2-patch3-8.4.0\xtensa-esp32s3-elf\bin;C:\Espressif\tools\riscv32-esp-elf\esp-2021r2-patch3-8.4.0\riscv32-esp-elf\bin;C:\Espressif\tools\esp32ulp-elf\2.28.51-esp-20191205\esp32ulp-elf-binutils\bin;C:\Espressif\tools\esp32s2ulp-elf\2.28.51-esp-20191205\esp32s2ulp-elf-binutils\bin;C:\Espressif\tools\cmake\3.20.3\bin;C:\Espressif\tools\openocd-esp32\v0.11.0-esp32-20211220\openocd-esp32\bin;C:\Espressif\tools\ninja\1.10.2\;C:\Espressif\tools\idf-exe\1.0.3\;C:\Espressif\tools\ccache\4.3\ccache-4.3-windows-64;C:\Espressif\tools\dfu-util\0.9\dfu-util-0.9-win64;C:\Espressif\frameworks\esp-idf-v4.4.1\tools;C:\Users\banuprakash.gedi\AppData\Local\Programs\Python\Python310\Scripts\;C:\Users\banuprakash.gedi\AppData\Local\Programs\Python\Python310\;C:\Program Files\Java\jdk-11\bin;C:\Users\banuprakash.gedi\AppData\Local\Programs\Git\cmd;C:\Program Files\apache-maven-3.8.6\bin;
      IDF_PATH: C:\Espressif\frameworks\esp-idf
   
    steps:
    - uses: actions/checkout@v3
    - name: build the imagefile
      run:  C:\Espressif\frameworks\esp-idf\install.bat
      shell: cmd
      working-directory: C:\Espressif\frameworks\esp-idf

![image](https://user-images.githubusercontent.com/89842780/190559218-c2a63303-3057-4c54-b51b-cb49bff747a3.png)

**Comment by naikrovek** [comment]
try this instead: `run:  cmd /c “C:\Espressif\frameworks\esp-idf\install.bat”`

**Reply by epneo-banuprakash** [reply]
![image](https://user-images.githubusercontent.com/89842780/190953019-de9e28d0-8c1d-435a-92f7-5109cb6d679f.png)


This is also not worked , and as u mentioned i changed the run command like below

![image](https://user-images.githubusercontent.com/89842780/190953174-7fd5a629-e379-417f-8839-6ae7b812fdbd.png)

**Reply by naikrovek** [reply]
well, I guess the PATH variable is not present or is wrong.

c:\windows\system32\cmd.exe instead of just "cmd" I guess.

**Reply by epneo-banuprakash** [reply]
Hi,
I have done some system settings , now its working like great.
Thanks for you attention.

**Reply by adityasomic** [reply]
Hi @naikrovek ,

If there is bat file in the github repository how to run that file as I am getting the errors. I have one file in the main repo which I would like to run with the shell as cmd 

      - name: Run the Script_Test_Manager
        run:  ./test_script.bat
        shell: cmd

but I am getting the error ' is not recognized as an internal or external command,
operable program or batch file.'

**Reply by urvideshpande** [reply]
Hi @adityasomic Did you get the resolution to your query? Are you able to run the .bat file which is there on git repository?

**Reply by adityasomic** [reply]
@urvideshpande yes I was able to run it

**Reply by urvi-deshpande** [reply]
@adityasomic May I know how did you run .bat file through GitHub actions workflow?

**Comment by chauhanhitendra** [comment]
> Hi, I have done some system settings , now its working like great. Thanks for you attention.
